From 57873d3896b67b5bb0e2f00127e39e3223987d2f Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Date: Wed, 13 Mar 2019 14:23:16 -0700
Subject: [PATCH 0898/1239] configs: Enable necessary configuration for MMC to
 work.

Some configurations were missed out in previous config files.

Change-Id: I5daaa0902fc7f8d4292567984312b0c04b13737b
Signed-off-by: Chandrakala Chavva <Chandrakala.Chavva@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/5654
Tested-by: sa_ip-sw-jenkins
---
 configs/octeontx2_95xx_defconfig |  2 --
 configs/octeontx2_96xx_defconfig |  5 +++--
 include/configs/octeontx2_95xx.h |  5 +++++
 include/configs/octeontx2_96xx.h | 11 ++++++++---
 4 files changed, 16 insertions(+), 7 deletions(-)

diff --git a/configs/octeontx2_95xx_defconfig b/configs/octeontx2_95xx_defconfig
index bd15b63708..4b60574309 100644
--- a/configs/octeontx2_95xx_defconfig
+++ b/configs/octeontx2_95xx_defconfig
@@ -48,7 +48,6 @@ CONFIG_OF_BOARD=y
 CONFIG_ENV_IS_IN_SPI_FLASH=y
 CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_BLOCK_CACHE=y
-CONFIG_BLK=y
 CONFIG_DM_GPIO=y
 CONFIG_I2C_GPIO=y
 CONFIG_DM_I2C=y
@@ -62,7 +61,6 @@ CONFIG_DM_MMC=y
 CONFIG_DM_REGULATOR=y
 CONFIG_DM_REGULATOR_FIXED=y
 CONFIG_DM_REGULATOR_GPIO=y
-CONFIG_MMC_OCTEONTX=y
 CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
diff --git a/configs/octeontx2_96xx_defconfig b/configs/octeontx2_96xx_defconfig
index 7f39f2542c..3bdcc78f73 100644
--- a/configs/octeontx2_96xx_defconfig
+++ b/configs/octeontx2_96xx_defconfig
@@ -58,7 +58,6 @@ CONFIG_NET_RANDOM_ETHADDR=y
 CONFIG_SCSI_AHCI=y
 CONFIG_AHCI_PCI=y
 CONFIG_BLOCK_CACHE=y
-CONFIG_BLK=y
 CONFIG_NVME=y
 CONFIG_DM_GPIO=y
 CONFIG_I2C_GPIO=y
@@ -69,7 +68,9 @@ CONFIG_I2C_MUX_PCA954x=y
 CONFIG_MISC=y
 CONFIG_MMC=y
 CONFIG_DM_MMC=y
-CONFIG_MMC_OCTEONTX=y
+CONFIG_DM_REGULATOR=y
+CONFIG_DM_REGULATOR_FIXED=y
+CONFIG_DM_REGULATOR_GPIO=y
 CONFIG_MTD=y
 CONFIG_DM_SPI_FLASH=y
 CONFIG_SPI_FLASH=y
diff --git a/include/configs/octeontx2_95xx.h b/include/configs/octeontx2_95xx.h
index 87aa5d4d9a..f439ffde19 100644
--- a/include/configs/octeontx2_95xx.h
+++ b/include/configs/octeontx2_95xx.h
@@ -101,4 +101,9 @@
 #define CONFIG_CMD_BKOPS_ENABLE
 #define CONFIG_SUPPORT_EMMC_RPMB
 
+#define CONFIG_MMC_OCTEONTX
+#ifndef CONFIG_BLK
+# define CONFIG_BLK
+#endif
+
 #endif /* __OCTEONTX2_95XX_H__ */
diff --git a/include/configs/octeontx2_96xx.h b/include/configs/octeontx2_96xx.h
index f713400ab5..3bc934dec7 100644
--- a/include/configs/octeontx2_96xx.h
+++ b/include/configs/octeontx2_96xx.h
@@ -107,9 +107,14 @@
 #undef CONFIG_SYS_PROMPT
 #define CONFIG_SYS_PROMPT		env_get("prompt")
 
-#define CONFIG_MMC_SUPPORTS_TUNING
-#define MMC_SUPPORTS_TUNING
-#define CONFIG_CMD_BKOPS_ENABLE
+/** EMMC specific defines */
+#define CONFIG_SUPPORT_EMMC_BOOT
 #define CONFIG_SUPPORT_EMMC_RPMB
+#define CONFIG_CMD_BKOPS_ENABLE
+
+#define CONFIG_MMC_OCTEONTX
+#ifndef CONFIG_BLK
+# define CONFIG_BLK
+#endif
 
 #endif /* __OCTEONTX2_96XX_H__ */
-- 
2.29.0

